### DISCUSSION TITLE
Azure Pipelines Scaler Exit Behavior (Azure Container Apps)

### DISCUSSION THREAD

**Discussion by jordanrbaker** [kedacore/keda#5715] [Q&A / Need Help]
I'm currently using https://keda.sh/docs/2.13/scalers/azure-pipelines/ to scale some self-hosted agents, and recently noticed they no longer appear to execute the cleanup function when the scaledown occurs.  I didn't see any changes in the ACA documentation that would indicate why this would be the case, and my github runners using similar code clean themselves up as expected.  Reaching out to see if anyone in the community has any ideas why they may not be cleaning up as expected.

Relevant details:

I set up a test using the baseline code here: https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops#linux.  Built and deployed to ACA.  Here's my observations:

- The agents work and execute pipelines as expected.  They clean themselves up appropriately after a single pipeline run as I'd expect with the `--once` flag.
- If I redeploy the ACA, the revision seems to exit gracefully and the cleanup() function is executed.
- If I STOP the ACA, the revisions do not do not cleanup (and this may be expected as I think stopping the app sends the kill command).
- If I kick off extra pipeline runs, extra revisions are created as expected.
- Once the revisions scale back down to one, the removed revisions do not clean up as expected.
- Nothing in the either the container or app logs to indicate why they wouldn't terminate as expected. 

I'm using Bicep to deploy, with the following KEDA scale setup to scale based on number of jobs in queue in the pool:

```
      scale: {
        minReplicas: minReplicas
        maxReplicas: maxReplicas
        rules: [
          {
            custom: {
                metadata: {
                  poolName: AZP_POOL
                  personalAccessTokenFromEnv: 'AZP_TOKEN'
                  organizationURLFromEnv: 'AZP_URL'
                  targetPipelinesQueueLength: '1'
              }
              type: 'azure-pipelines'
            }
            name: 'adoscalingrule'          
          }
        ]
      }
    }
  }
}
```

**Comment by tomkerkhove** [comment]
Hi, for Azure Container Apps we ask people to file an issue on https://github.com/microsoft/azure-container-apps/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc.

This repo is only for open-source KEDA.
